### PR TITLE
Traitor Vamp

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Adds in TraitorVamp--it's essentially TraitorChan, but instead of a changeling, it's a vampire.

Related
- Vampires cannot use holoparasites, much like cahngelings

Other
- Reduces requirement for this game mode to activate

:cl: Fox McCloud
add: Adds in TraitorVamp; game mode that's essentially like TraitorChan, but with a Vampire instead of a Changeling
tweak: Vampires cannot use holoparasites
/:cl:

### CODE REVIEW

**Comment by FalseIncarnate** [comment]
Vampires with traitor gear is sure going to be interesting to see.

**Comment by Necaladun** [comment]
Yayyy. More screwing with meta!

**Comment by Aurorablade** [comment]
request: QoL for voxpires....you know my suggestion for it...

**Comment by KasparoVy** [comment]
@Aurorablade Why not post it here for posterity?

**Comment by Fox-McCloud** [comment]
Got around to testing this; seems to work fine---also made the layout for enthralled a bit better/more clear for admins.

File: code/datums/mind.dm [hunk 1]
```diff
@@ -257,7 +257,7 @@
 
 		/** VAMPIRE ***/
 		text = "vampire"
-		if (ticker.mode.config_tag=="vampire")
+		if (ticker.mode.config_tag=="vampire" || ticker.mode.config_tag=="traitorvamp")
```
Comment (tigercat2000) [inline code/datums/mind.dm]: # GAMEMODE CODE EVERYONE!
Isn't it fucking ~~atrocious~~ ~~blatantly ignoring all object oriented programming techniques~~ BEAUTIFUL?

File: code/game/gamemodes/changeling/traitor_chan.dm [hunk 2]
```diff
@@ -3,9 +3,9 @@
 	config_tag = "traitorchan"
 	traitors_possible = 3 //hard limit on traitors if scaling is turned off
 	restricted_jobs = list("AI", "Cyborg")
-	required_players = 3
-	required_players_secret = 10
-	required_enemies = 2
+	required_players = 0
+	required_players_secret = 0
```
Comment (tigercat2000) [inline code/game/gamemodes/changeling/traitor_chan.dm]: why

File: code/datums/mind.dm (hunk 1 above)
Comment (Spacemanspark) [inline code/datums/mind.dm]: Yes.

File: code/game/gamemodes/changeling/traitor_chan.dm (hunk 2 above)
Comment (Tastyfish) [inline code/game/gamemodes/changeling/traitor_chan.dm]: Was this a debugging thing left in?

File: code/game/gamemodes/vampire/traitor_vamp.dm [hunk 3]
```diff
@@ -0,0 +1,45 @@
+/datum/game_mode/traitor/vampire
+	name = "traitor+vampire"
+	config_tag = "traitorvamp"
+	traitors_possible = 3 //hard limit on traitors if scaling is turned off
+	restricted_jobs = list("AI", "Cyborg")
+	required_players = 0
+	required_players_secret = 0
```
Comment (Tastyfish) [inline code/game/gamemodes/vampire/traitor_vamp.dm]: Also here too, pretty sure the game mode would be terrible with only 1 or 2 players.

**Comment by TheDZD** [comment]
Hooray, less boring vampire rounds. On the downside, there's going to be hardly any antags relative to station population because traitorchan and now traitorvamp will not scale antags off of late-joining players, which have a tendency to get very close to, or exceed the number of players ready at round start.

**Comment by Necaladun** [comment]
Autovamp/autoling would be great, at least for late-joins. "I can't be a target I joined late" shits me to no end.

**Comment by Mijhead** [comment]
No holoparasites for vampires?

**Pull request merged**